### PR TITLE
fix(Message#editable): update editable check in threads locked

### PR DESCRIPTION
**Pull request by ErwanGit**
**Please describe the changes this PR makes and why it should be merged:**

[Recent changes](https://discord.com/developers/docs/change-log#update-to-locked-threads) regarding locked threads were rolled out this week.
If a bot does not have permission to manage threads, it will not be able to edit its own message even if the thread is not archived.
I've also added a check that was probably forgotten, a bot can't edit its own message if the thread is archived.

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating

### CODE REVIEW

**Comment by jaw0r3k** [comment]
what? is doesn't make any sense
1. you deleted checking precheck
2. as I understood well locked still means locked
so you can edit messages in archived thread but not in locked 
3. only thing to leave is checking perms

**Comment by jaw0r3k** [comment]
yeah the logic doesn't make any sense
as even in thread isn't archived it will check for the permissions

**Comment by ErwanGit** [comment]
I re-tested before submitting this PR and concluded that : 
- if the thread is archived : impossible to modify the message without unarchiving the thread
- if the thread is locked : it is possible to modify the message only if the bot has the permission to manage the threads in the channel

I could make changes to check permissions only if the thread is locked, this is a mistake on my part 
But after that, I don't see what I could change, the current behavior of the editable check on locked threads is not good at the moment

File: packages/discord.js/src/structures/Message.js [hunk 1]
```diff
@@ -602,11 +602,17 @@ class Message extends Base {
    */
   get editable() {
     const precheck = Boolean(this.author.id === this.client.user.id && (!this.guild || this.channel?.viewable));
+
     // Regardless of permissions thread messages cannot be edited if
-    // the thread is locked.
+    // the thread is archived or the thread is locked and the bot does not have permission to manage threads
```
Comment (jaw0r3k) [inline packages/discord.js/src/structures/Message.js:607]:
```suggestion
    // the thread is archived or the thread is locked and the bot does not have permission to manage threads.
```

**Review by DraftProducts — APPROVED** [review]

**Comment by jaw0r3k** [comment]
still, you can edit messages in archived channel....

**Comment by ErwanGit** [comment]
> still, you can edit messages in archived channel....

I just tried again and it seems that it is not possible to edit a post in an archived thread
image

**Review by SpaceEEC — APPROVED** [review]

**Review by vladfrangu — APPROVED** [review]

**Pull request merged**